lib/main.dart

```diff
@@ -42,23 +43,79 @@ class MyAppState extends State<MyApp> {
   Widget build(BuildContext context) {
     return GlobalProvider(
       globalBloc: globalBloc,
-      child: MaterialApp(
-        title: 'EventsFlutter',
-        debugShowCheckedModeBanner: false,
-        theme: ThemeData(
-            primarySwatch: Colors.red, primaryColor: Color(0xffB54646)),
-        home: StreamBuilder(
-          // HUB state builder
-          stream: globalBloc.hubStateStreamController.stream,
-          initialData: ShowSplashState(),
-          builder: (context, snapshot) {
-            if (snapshot.data is ShowMainState) {
-              globalBloc.disposeSplashController();
-              return MainScreen();
-            }
-            return SplashScreen();
-          },
-        ),
+      child: StreamBuilder(
```

Comment:
There is no need for a StreamBuilder here. Because this is a `StatefulWidget`.
Instead, you can do something like this - (pseudo code)
```dart
void initState(){
  ...
  themeStream.listen((){
      // Call setstate here
      if(widget.mounted){
      setState((){});
  }
  });
}
```